beacon-chain/powchain/log_processing.go

```diff
@@ -411,6 +415,7 @@ func (s *Service) requestBatchedHeadersAndLogs(ctx context.Context) error {
 		log.Infof("Falling back to historical headers and logs sync. Current difference is %d", requestedBlock-s.latestEth1Data.LastRequestedBlock)
 		return s.processPastLogs(ctx)
 	}
+	s.latestEth1DataLock.Lock()
```

Comment:
This acquires the lock for the whole length of the loop, if any of the inner methods take longer than usual, other methods can be blocked.